network/endpoint_linux.go

```diff
@@ -437,12 +437,12 @@ func updateRoutes(existingEp *EndpointInfo, targetEp *EndpointInfo) error {
 
 	}
 
-	err := deleteRoutes(existingEp.IfName, tobeDeletedRoutes)
+	err := deleteRoutes(nl, existingEp.IfName, tobeDeletedRoutes)
 	if err != nil {
 		return err
 	}
 
-	err = addRoutes(existingEp.IfName, tobeAddedRoutes)
+	err = addRoutes(nl, existingEp.IfName, tobeAddedRoutes)
```

Comment:
addroutes and deleteroutes can have networkmanager as objects

Reply:
Keeping this simple by passing in the netlink object and not the whole NM object